packages/app/src/App.tsx

```diff
@@ -48,17 +51,39 @@ const AppComponent = app.build();
 
 const App: FC<{}> = () => {
   useStyles();
+  let themeId : string = localStorage.getItem('light') || 'auto';
+  if (['light', 'dark'].indexOf(themeId) < 0) {
+    themeId = useMediaQuery('(prefers-color-scheme: dark)') ? 'dark' : 'light';
+  }
+  const [theme, setTheme] = useState(themeId);
+  const toggleTheme = () => {
+    if (theme === 'light') {
+      setTheme('dark');
+      localStorage.setItem('light', 'dark');
+    } else if (theme === 'light') {
+      setTheme('auto');
+      localStorage.setItem('light', 'auto');
+      themeId = useMediaQuery('(prefers-color-scheme: dark)') ? 'dark' : 'light';
+    } else {
+      setTheme('light');
+      localStorage.setItem('light', 'light');
+    }
+  }
+  const backstageThemeDark = BackstageThemeDark;
+  const backstageThemeLight = BackstageThemeLight;
   return (
-    <CssBaseline>
-      <ThemeProvider theme={BackstageTheme}>
-        <ErrorDisplay forwarder={errorDialogForwarder} />
-        <Router>
-          <Root>
-            <AppComponent />
-          </Root>
-        </Router>
+    <ThemeContext.Provider value={{theme, toggleTheme}}>
+      <ThemeProvider theme={theme === 'dark' ? backstageThemeDark : backstageThemeLight}>
+        <CssBaseline>
```

Comment:
Hm! Is the baseline intentionally inside the theme? I thought it needed to be the other way around

Reply:
Yes it is needed to be in order t o work